ddtrace/tracer/tracer.go

```diff
@@ -57,6 +60,36 @@ type tracer struct {
 	pid string
 }
 
+type statsdClient interface {
+	Incr(name string, tags []string, rate float64) error
+	Count(name string, value int64, tags []string, rate float64) error
+	Gauge(name string, value float64, tags []string, rate float64) error
+	Timing(name string, value time.Duration, tags []string, rate float64) error
+	Close() error
+}
+
+type noopStats struct{}
```

Comment:
I'm not sure this name is descriptive of what this is. Can we use `noopStatsdClient`?